File: javascript/7kyu/Thinking & Testing Something capitalized.js
// https://www.codewars.com/kata/56d93f249c844788bc000002

function testit(s){
    let str = '';
    for (i = 0; i < s.length; i++) {
       if (s[i + 1] === ' ' || i === s.length - 1) {
          str += s[i].toUpperCase();
       } else {
          str += s[i]
       }
    }
    return str;
  }

File: java/7kyu/Thinking & Testing Something capitalized.java
// https://www.codewars.com/kata/56d93f249c844788bc000002

import java.util.stream.Collectors;
import java.util.Arrays;

public class ThinkingAndTesting {
  public static String testSomethingCapitalized(String s) {
    return Arrays.stream(s.split(" "))
                .map(x -> x.length() > 0 ? x.substring(0, x.length()-1) + x.substring(x.length()-1).toUpperCase() : x)
                .collect(Collectors.joining(" "));
  }
}